feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/domain/referendum/vote/RealVoteReferendumInteractor.kt

```diff
@@ -60,40 +46,58 @@ class RealVoteReferendumInteractor(
 ) : VoteReferendumInteractor {
 
     override fun voteAssistantFlow(referendumId: ReferendumId, scope: CoroutineScope): Flow<GovernanceVoteAssistant> {
+        return voteAssistantFlow(listOf(referendumId), scope)
+    }
+
+    override fun voteAssistantFlow(referendaIds: List<ReferendumId>, scope: CoroutineScope): Flow<GovernanceVoteAssistant> {
         return computationalCache.useSharedFlow(VOTE_ASSISTANT_CACHE_KEY, scope) {
             val governanceOption = selectedChainState.selectedOption()
             val metaAccount = accountRepository.getSelectedMetaAccount()
 
             val voterAccountId = metaAccount.accountIdIn(governanceOption.assetWithChain.chain)!!
 
-            voteAssistantFlowSuspend(governanceOption, voterAccountId, metaAccount.id, referendumId)
+            voteAssistantFlowSuspend(governanceOption, voterAccountId, metaAccount.id, referendaIds)
         }
     }
 
-    override suspend fun estimateFee(amount: Balance, conviction: Conviction, referendumId: ReferendumId): Fee {
+    override suspend fun estimateFee(votes: Map<ReferendumId, AccountVote>): Fee {
+        return estimateFeeInternal { it.vote(votes) }
+    }
+
+    override suspend fun estimateFee(referendumId: ReferendumId, vote: AccountVote): Fee {
+        return estimateFeeInternal { it.vote(referendumId, vote) }
+    }
+
+    private suspend inline fun estimateFeeInternal(crossinline builder: ConvictionVotingRepository.(ExtrinsicBuilder) -> Unit): Fee {
         val governanceOption = selectedChainState.selectedOption()
         val chain = governanceOption.assetWithChain.chain
 
-        val vote = AyeVote(amount, conviction) // vote direction does not influence fee estimation
         val governanceSource = governanceSourceRegistry.sourceFor(governanceOption)
 
         return extrinsicService.estimateFee(chain, TransactionOrigin.SelectedWallet) {
             with(governanceSource.convictionVoting) {
-                vote(referendumId, vote)
+                builder(this@estimateFee)
             }
         }
     }
 
-    override suspend fun vote(
-        vote: AccountVote,
-        referendumId: ReferendumId,
-    ): Result<ExtrinsicSubmission> {
-        val governanceSelectedOption = selectedChainState.selectedOption()
-        val governanceSource = governanceSourceRegistry.sourceFor(governanceSelectedOption)
+    override suspend fun vote(votes: Map<ReferendumId, AccountVote>): Result<ExtrinsicSubmission> {
+        return voteInternal { it.vote(votes) }
```

Comment:
For SwipeGov, we might not be able to fit all votes into a single block so we should use `submitMultiExtrinsicAwaitingInclusion`